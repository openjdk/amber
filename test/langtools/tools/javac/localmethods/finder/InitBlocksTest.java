/**
 * @test /nodynamiccopyright/
 * @bug 8233754
 * @compile/fail/ref=InitBlocksTest.out -Werror -XDrawDiagnostics -XDfind=local-methods InitBlocksTest.java
 */

public class InitBlocksTest {

    private static void goo() {}  // can't be localized
    private static void soo() {}  // can't be localized
    private static void boo() {}  // can't be localized
    private static void loo() {}  // can be 

    private void goo(int i) {}    // can't be localized
    private void soo(int i) {}    // can't be localized
    private void loo(int i) {}    // can be

    static {
        goo();
        loo();
        boo();
    }

    static {
        goo();
        soo();
    }

    {
        goo(42);
        loo(42);
        boo();
    }

    {
        goo(42);
        soo(42);
    }

    public static void main(String [] args) {
        soo();
    }
    public void main() {
        soo(42);
    }
}
