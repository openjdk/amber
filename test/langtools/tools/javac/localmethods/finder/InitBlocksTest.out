InitBlocksTest.java:12:25: compiler.warn.method.could.be.declared.local.to.static.initializer: kindname.class, InitBlocksTest
InitBlocksTest.java:16:18: compiler.warn.method.could.be.declared.local.to.instance.initializer: kindname.class, InitBlocksTest
- compiler.err.warnings.and.werror
1 error
2 warnings
